backend/src/api/model/realm/mutations.rs

```diff
@@ -166,6 +173,31 @@ impl Realm {
         let key = id_to_key(id, "`id`")?;
         let parent_key = set.parent.map(|parent| id_to_key(parent, "`parent`")).transpose()?;
 
+        // We have to make sure the path is not changed to a reserved one.
```

Comment:
Again, I find the order here unintuitive. First checking all the reserved paths and only then seeing whether it's even necessary. :thinking:

Reply:
Well. Checking `parent_key == Some(0)` is super fast, checking all reserved paths (which are only a handful in practice) takes a bit longer, talking to the DB is by far the longest operation. So we want to avoid the database talking at all cost. The other two... whatever.

Reply:
Unfortunate but fair. :/